stdlib/arrayLabels.mli

```diff
@@ -202,6 +202,16 @@ val for_all : f:('a -> bool) -> 'a array -> bool
    [(f a1) && (f a2) && ... && (f an)].
    @since 4.03.0 *)
 
+val for_all2 : f:('a -> 'b -> bool) -> 'a array -> 'b array -> bool
+(** Same as {!ArrayLabels.for_all}, but for a two-argument predicate.
+   Raise [Invalid_argument] if the two arrays are determined
+   to have different lengths. *)
```

Comment:
Since the failure in the case of different lengths is immediate (instead of occurring after some possibly complicated computation), I would simplify the phrasing and simply say "if the two arrays have different lengths". 

To me, the "are determined to have" phrasing suggests that some non trivial computation might occur before raising the error---which is the case for lists, but not for arrays as you point out.

Reply:
Thanks; fixed (a6dc51b).